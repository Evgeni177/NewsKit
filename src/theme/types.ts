import {StylePreset, TypographyPreset} from './presets/types';
import {DeepPartial} from '../utils/types';
import {colors} from './foundations';

export type Breakpoints = Record<'xs' | 'sm' | 'md' | 'lg' | 'xl', number>;
export enum Devices {
  iPad = 'iPad',
  iPadPro = 'iPad Pro',
}

export interface ThemeBase {
  // foundations
  motions: Record<string, string>;
  borders: Record<string, string>;
  breakpoints: Breakpoints;
  colors: Record<string, string>;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  fonts: Record<string, any>;
  overlays: Record<string, string>;
  shadows: Record<string, string>;
  sizing: Record<string, string>;

  // presets
  spacePresets: Record<string, string>;
  stylePresets: Record<string, StylePreset>;
  typographyPresets: Record<string, TypographyPreset>;

  // defaults
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  componentDefaults: Record<string, any>;
  icons: Record<string, React.ComponentType>;
}

export type ThemeOverrides = DeepPartial<ThemeBase>;

interface ThemeIdentifier {
  themeVersion: 1;
  name: string;
}

export interface UncompiledTheme extends ThemeBase, ThemeIdentifier {
  compiled?: false;
}

export interface Theme extends ThemeBase, ThemeIdentifier {
  compiled: true;
}

export type MotionKeys = string;
export type BorderKeys = string;
export type BorderRadiusKeys = string;
export type BreakpointKeys = keyof Breakpoints;
export type ColorKeys = string;
export type IconSizeKeys = string;
export type SizingKeys = string;
export type StylePresetKeys = string;
export type TypographyPresetKeys = string;
export type SpacePresetKeys = string;
export type PaddingPresetKeys = string;
export type ShadowKeys = string;
export type GridKeys = string;
export type FontKeys = string;
export type FontSizeKeys = string;
export type LineHeightKeys = string;
export type Colors = typeof colors;

export * from './presets/types';
