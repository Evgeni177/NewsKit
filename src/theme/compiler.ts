/* eslint-disable no-use-before-define */
/* eslint-disable @typescript-eslint/no-explicit-any */
import {Theme, UncompiledTheme} from './types';
import {get} from '../utils/get';

const parseTokens = (str: unknown) => {
  if (typeof str !== 'string') {
    return [];
  }

  const result = [];
  let res: RegExpExecArray | null;

  const regex = /{{([a-z0-9.]+)}}/gi;
  // eslint-disable-next-line no-cond-assign
  while ((res = regex.exec(str))) {
    result.push(res[1]);
  }

  return result;
};

const parseAndGet = (
  theme: UncompiledTheme,
  value: unknown,
  stack: unknown[] = [],
): unknown => {
  if (stack.length > 4) {
    // eslint-disable-next-line no-console
    console.error(
      `Recursive loop detected, token stack: "${stack.join('", "')}"!`,
    );
    return '';
  }

  const tokens = parseTokens(value);
  if (tokens.length) {
    return tokens.reduce(
      (result: any, tokenPath) => {
        let tokenValue = get(theme, tokenPath);
        if (typeof tokenValue === 'undefined') {
          // eslint-disable-next-line no-console
          console.error(
            `Theme token "${tokenPath}" was not found when compiling theme!`,
          );
        } else {
          // We recurse down with that token value to support things like colors
          // e.g. (border color = semanticNegative010 = red010 = #ff0000).
          tokenValue = parseAndGet(theme, tokenValue, [tokenPath, ...stack]);
        }

        return tokens.length > 1
          ? result.replace(`{{${tokenPath}}}`, tokenValue as string)
          : recurseUnknown(theme, tokenValue);
      },
      value as string,
    );
  }

  // Not a token, return as is (e.g. hex code or px value).
  return value;
};

const recurseUnknown = (theme: UncompiledTheme, value: unknown): unknown => {
  if (typeof value === 'function') {
    // Evaluation function, e.g. for calculating line height, run function and pass it the theme.
    const fResult = value(theme);
    // Result could be an object (with tokens), so recursively compile.
    return recurseUnknown(theme, fResult);
  }
  if (typeof value === 'object' && !Array.isArray(value)) {
    // Nested object, recurse down
    return Object.entries(value as any).reduce(
      (acc, [k, v]: any) => {
        // Key could be a token, e.g. cropAdjustments fonts object
        const key = parseAndGet(theme, k) as string;
        acc[key] = recurseUnknown(theme, v);
        return acc;
      },
      {} as any,
    );
  }
  // Anything else, probably string, parse and get the token value
  return parseAndGet(theme, value);
};

export const compileTheme = (theme: UncompiledTheme | Theme): Theme => {
  if (!theme || theme.compiled === true) {
    return theme;
  }

  const {icons = {}, ...filteredTheme} = theme;
  return {
    ...(recurseUnknown(theme, filteredTheme) as any),
    icons,
    compiled: true,
  };
};
