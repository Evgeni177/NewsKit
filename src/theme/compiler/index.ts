/* eslint-disable no-use-before-define */
/* eslint-disable @typescript-eslint/no-explicit-any */
import {ThemeLoggerFunction, Theme, UncompiledTheme} from '../types';
import {get} from '../../utils/get';
import {ThemeCompilerOptions} from './types';

const parseTokens = (str: unknown) => {
  if (typeof str !== 'string') {
    return [];
  }

  const result = [];
  let res: RegExpExecArray | null;

  const regex = /{{([a-z0-9.]+)}}/gi;
  // eslint-disable-next-line no-cond-assign
  while ((res = regex.exec(str))) {
    result.push(res[1]);
  }

  return result;
};

const parseAndGet = (
  theme: UncompiledTheme,
  value: unknown,
  errorLogger: ThemeLoggerFunction,
  stack: unknown[] = [],
): unknown => {
  if (stack.length > 4) {
    errorLogger(
      `Recursive loop detected, token stack: "${stack.join('", "')}"!`,
    );
    return '';
  }

  const tokens = parseTokens(value);
  if (tokens.length) {
    return tokens.reduce(
      (result: any, tokenPath) => {
        let tokenValue = get(theme, tokenPath);
        if (typeof tokenValue === 'undefined') {
          errorLogger(
            `Theme token "${tokenPath}" was not found when compiling theme!`,
          );
        } else {
          // We recurse down with that token value to support things like colors
          // e.g. (border color = interactiveNegative010 = red010 = #ff0000).
          tokenValue = parseAndGet(theme, tokenValue, errorLogger, [
            tokenPath,
            ...stack,
          ]);
        }

        return tokens.length > 1
          ? result.replace(`{{${tokenPath}}}`, tokenValue as string)
          : recurseUnknown(theme, tokenValue, errorLogger);
      },
      value as string,
    );
  }

  // Not a token, return as is (e.g. hex code or px value).
  return value;
};

const recurseUnknown = (
  theme: UncompiledTheme,
  value: unknown,
  errorLogger: ThemeLoggerFunction,
): unknown => {
  if (typeof value === 'function') {
    // Evaluation function, e.g. for calculating line height, run function and pass it the theme.
    const fResult = value(theme);
    // Result could be an object (with tokens), so recursively compile.
    return recurseUnknown(theme, fResult, errorLogger);
  }
  if (typeof value === 'object' && !Array.isArray(value)) {
    // Nested object, recurse down
    return Object.entries(value as any).reduce(
      (acc, [k, v]: any) => {
        // Key could be a token, e.g. cropAdjustments fonts object
        const key = parseAndGet(theme, k, errorLogger) as string;
        acc[key] = recurseUnknown(theme, v, errorLogger);
        return acc;
      },
      {} as any,
    );
  }
  // Anything else, probably string, parse and get the token value
  return parseAndGet(theme, value, errorLogger);
};

export const compileTheme = (
  theme: UncompiledTheme | Theme,
  options: ThemeCompilerOptions = {},
): Theme => {
  if (!theme || theme.compiled === true) {
    return theme;
  }

  // eslint-disable-next-line no-console
  const errorLogger = options.errorLogger || console.error.bind(console);

  const {icons = {}, ...filteredTheme} = theme;
  return {
    ...(recurseUnknown(theme, filteredTheme, errorLogger) as any),
    icons,
    compiled: true,
  };
};
