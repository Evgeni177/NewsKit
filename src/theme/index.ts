export * from './compiler';
export * from './creator';
export * from './emotion';
export * from './newskit-dark';
export * from './newskit-light';
export * from './types';
