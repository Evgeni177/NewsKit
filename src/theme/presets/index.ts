export * from './space-presets';
export * from './style-presets';
export * from './typography-presets';
export * from './transition-presets';
export * from './types';
