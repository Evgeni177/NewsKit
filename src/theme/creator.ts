import {deepMerge} from '../utils/deep-merge';

import {ThemeOverrides, UncompiledTheme, ThemeBase} from './types';

import * as foundations from './foundations';
import * as presets from './presets';

import {componentDefaults} from './component-defaults';

interface CreateThemeArgs {
  name?: string;
  baseTheme?: UncompiledTheme;
  overrides?: ThemeOverrides;
}

export const createTheme = ({
  name = 'unnamed-newskit-theme',
  baseTheme,
  overrides = {},
}: CreateThemeArgs): UncompiledTheme => {
  if (baseTheme && baseTheme.compiled) {
    throw new Error(
      'createTheme received a compiled baseTheme. Base themes must be uncompiled.',
    );
  }
  return deepMerge(
    {
      themeVersion: 1,
      ...foundations,
      ...presets,
      componentDefaults,
      icons: {},
    } as ThemeBase,
    baseTheme,
    overrides,
    {name},
  );
};
