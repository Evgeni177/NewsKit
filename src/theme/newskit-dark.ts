import {createTheme} from './creator';
import {ThemeOverrides} from './types';

const overrides: ThemeOverrides = {
  colors: {
    inkBase: '{{colors.neutral020}}',
    inkContrast: '{{colors.neutral010}}',
    inkSubtle: '{{colors.neutral040}}',
    inkNonEssential: '{{colors.neutral070}}',
    inkInverse: '{{colors.black}}',
    inkHeadingLink: '{{colors.neutral020}}',
    inkHeadingLinkHover: '{{colors.blue020}}',
    inkHeadingLinkActive: '{{colors.blue020}}',
    inkHeadingLinkVisited: '{{colors.neutral020}}',
    inkNegative: '{{colors.red030}}',
    inkPositive: '{{colors.green030}}',
    inkNotice: '{{colors.amber040}}',
    inkInformative: '{{colors.purple020}}',
    inkBrand010: '{{colors.blue040}}',
    inkLink: '{{colors.blue020}}',
    inkLinkHover: '{{colors.blue020}}',
    inkLinkActive: '{{colors.blue020}}',
    inkLinkVisited: '{{colors.purple020}}',
    semanticNegative010: '{{colors.red090}}',
    semanticNegative020: '{{colors.red100}}',
    semanticNegative030: '{{colors.red060}}',
    semanticNegative040: '{{colors.red070}}',
    semanticNegative050: '{{colors.red080}}',
    semanticPositive010: '{{colors.green090}}',
    semanticPositive020: '{{colors.green100}}',
    semanticPositive030: '{{colors.green060}}',
    semanticPositive040: '{{colors.green070}}',
    semanticPositive050: '{{colors.green080}}',
    semanticInformative010: '{{colors.purple040}}',
    semanticNotice010: '{{colors.amber030}}',
    interfaceBackground: '{{colors.neutral090}}',
    interface010: '{{colors.neutral090}}',
    interface020: '{{colors.neutral080}}',
    interface030: '{{colors.neutral070}}',
    interface040: '{{colors.neutral060}}',
    interface050: '{{colors.neutral050}}',
    interface060: '{{colors.neutral010}}',
    interfaceBrand010: '{{colors.blue060}}',
    interfaceBrand020: '{{colors.blue020}}',
    interactive010: '{{colors.blue080}}',
    interactive020: '{{colors.blue090}}',
    interactive030: '{{colors.blue060}}',
    interactive040: '{{colors.blue070}}',
    interactive050: '{{colors.blue080}}',
    interactive110: '{{colors.neutral080}}',
    interactive120: '{{colors.neutral090}}',
    interactive130: '{{colors.neutral070}}',
    interactive140: '{{colors.neutral080}}',
    interactive150: '{{colors.neutral090}}',
    interactive210: '{{colors.white}}',
    interactive220: '{{colors.neutral090}}',
    interactive230: '{{colors.blue060}}',
    interactive240: '{{colors.blue090}}',
    interactive250: '{{colors.blue100}}',
    interactiveDisabled: '{{colors.neutral090}}',
    disabled: '{{colors.neutral090}}',
    skeleton010: '{{colors.neutral090}}',
    skeleton020: '{{colors.neutral080}}',
    inverse: '{{colors.black}}',
  },
  overlays: {
    overlayLight010: 'rgba(255,255,255,0.20)',
    overlayLight020: 'rgba(255,255,255,0.40)',
    overlayLight030: 'rgba(255,255,255,0.60)',
    overlayLight040: 'rgba(255,255,255,0.80)',
    overlayDark010: 'rgba(10,10,10,0.20)',
    overlayDark020: 'rgba(10,10,10,0.40)',
    overlayDark030: 'rgba(10,10,10,0.60)',
    overlayDark040: 'rgba(10,10,10,0.80)',
    gradientLight010:
      'linear-gradient(180deg, rgba(255, 255, 255, 0) 0%, #FFFFFF 100%)',
    gradientDark010:
      'linear-gradient(180deg, rgba(51, 51, 51, 0) 0%, #0A0A0A 100%)',
  },
};

export const newskitDarkTheme = createTheme({
  name: 'newskit-dark',
  overrides,
});
