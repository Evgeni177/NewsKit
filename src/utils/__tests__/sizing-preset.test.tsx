import {renderToFragmentWithTheme} from '../../test/test-utils';
import {
  getMarginPresetFromTheme,
  getSpacingInsetFromTheme,
  styled,
  MQ,
} from '../style';
import {SpacePresetKeys, PaddingPresetKeys} from '../../theme';

interface SpacingPresetProp {
  margin?: MQ<SpacePresetKeys>;
  padding?: MQ<PaddingPresetKeys>;
}

const TestBlock = styled.div<SpacingPresetProp>`
  ${getMarginPresetFromTheme({sm: 'space040', md: 'space020'}, 'margin')}
  ${getSpacingInsetFromTheme(undefined, 'padding')}
`;

describe('MarginPreset', () => {
  test('renders default', () => {
    const fragment = renderToFragmentWithTheme(TestBlock);
    expect(fragment).toMatchSnapshot();
  });

  test('renders with override spaceStack010', () => {
    const fragment = renderToFragmentWithTheme(TestBlock, {
      margin: 'spaceStack010',
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders with override incorrect object', () => {
    const fragment = renderToFragmentWithTheme(TestBlock, {
      margin: {
        foo: 'bar',
      } as any,
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders responsive override spaceStack010 for xs, spaceStack020 for sm, and spaceStack030 for md breakpoints', () => {
    const fragment = renderToFragmentWithTheme(TestBlock, {
      margin: {
        xs: 'spaceStack010',
        sm: 'spaceStack020',
        md: 'spaceStack030',
      },
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders responsive override spaceStack010 for xs, spaceStack020 for sm, and wrong for md breakpoints', () => {
    const fragment = renderToFragmentWithTheme(TestBlock, {
      margin: {
        xs: 'spaceStack010',
        sm: 'spaceStack020',
        md: 'wrong',
      } as any,
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders responsive override spaceStack010 for sm and spaceStack030 for md breakpoints', () => {
    const fragment = renderToFragmentWithTheme(TestBlock, {
      margin: {
        sm: 'spaceStack010',
        md: 'spaceStack030',
      },
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders responsive override spaceStack010 for xs, spaceStack020 for sm, spaceStack030 for md, spaceStack040 for lg breakpoings, and ignore wrong prop', () => {
    const fragment = renderToFragmentWithTheme(TestBlock, {
      margin: {
        xs: 'spaceStack010',
        sm: 'spaceStack020',
        md: 'spaceStack030',
        lg: 'spaceStack040',
        wrong: 'spaceStack050',
      } as any,
    });
    expect(fragment).toMatchSnapshot();
  });
});

describe('SpaceInset', () => {
  test('renders default', () => {
    const fragment = renderToFragmentWithTheme(TestBlock);
    expect(fragment).toMatchSnapshot();
  });

  test('renders with override spaceInset010Squish', () => {
    const fragment = renderToFragmentWithTheme(TestBlock, {
      padding: 'spaceInset010Squish',
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders with override incorrect object', () => {
    const fragment = renderToFragmentWithTheme(TestBlock, {
      padding: {
        foo: 'bar',
      } as any,
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders responsive override spaceInset010Squish for xs, spaceInset020Squish for sm, and spaceInset030Squish for md breakpoints', () => {
    const fragment = renderToFragmentWithTheme(TestBlock, {
      padding: {
        xs: 'spaceInset010Squish',
        sm: 'spaceInset020Squish',
        md: 'spaceInset030Squish',
      },
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders responsive override spaceInset010Squish for xs, spaceInset020Squish for sm, and spaceInset030Squish for md breakpoints in reverser order', () => {
    const fragment = renderToFragmentWithTheme(TestBlock, {
      padding: {
        md: 'spaceInset030Squish',
        sm: 'spaceInset020Squish',
        xs: 'spaceInset010Squish',
      },
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders responsive override spaceInset010Squish for xs, spaceInset020Squish for sm, and wrong for md breakpoints', () => {
    const fragment = renderToFragmentWithTheme(TestBlock, {
      padding: {
        xs: 'spaceInset010Squish',
        sm: 'spaceInset020Squish',
        md: 'wrong',
      } as any,
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders responsive override spaceInset010Squish for xs, and spaceInset030Squish for md breakpoints', () => {
    const fragment = renderToFragmentWithTheme(TestBlock, {
      padding: {
        xs: 'spaceInset010Squish',
        md: 'spaceInset030Squish',
      },
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders responsive override spaceInset010Squish for xs, spaceInset020Squish for sm, spaceInset030Squish for md, spaceInset040Squish for lg breakpoints and ignore wrong prop', () => {
    const fragment = renderToFragmentWithTheme(TestBlock, {
      padding: {
        xs: 'spaceInset010Squish',
        sm: 'spaceInset020Squish',
        md: 'spaceInset030Squish',
        lg: 'spaceInset040Squish',
        wrong: 'spaceInset050Squish',
      } as any,
    });
    expect(fragment).toMatchSnapshot();
  });
});
