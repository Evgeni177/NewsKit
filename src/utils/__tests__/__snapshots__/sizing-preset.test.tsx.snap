// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarginPreset renders default 1`] = `
<DocumentFragment>
  @media screen and (min-width:480px) and (max-width:959px) {
  .emotion-0 {
    margin: 16px;
  }
}

@media screen and (min-width:960px) {
  .emotion-0 {
    margin: 8px;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`MarginPreset renders responsive override spaceStack010 for sm and spaceStack030 for md breakpoints 1`] = `
<DocumentFragment>
  <div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`MarginPreset renders responsive override spaceStack010 for xs, spaceStack020 for sm, and spaceStack030 for md breakpoints 1`] = `
<DocumentFragment>
  <div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`MarginPreset renders responsive override spaceStack010 for xs, spaceStack020 for sm, and wrong for md breakpoints 1`] = `
<DocumentFragment>
  <div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`MarginPreset renders responsive override spaceStack010 for xs, spaceStack020 for sm, spaceStack030 for md, spaceStack040 for lg breakpoings, and ignore wrong prop 1`] = `
<DocumentFragment>
  <div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`MarginPreset renders with override incorrect object 1`] = `
<DocumentFragment>
  <div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`MarginPreset renders with override spaceStack010 1`] = `
<DocumentFragment>
  <div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`SpaceInset renders default 1`] = `
<DocumentFragment>
  @media screen and (min-width:480px) and (max-width:959px) {
  .emotion-0 {
    margin: 16px;
  }
}

@media screen and (min-width:960px) {
  .emotion-0 {
    margin: 8px;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`SpaceInset renders responsive override spaceInset010Squish for xs, and spaceInset030Squish for md breakpoints 1`] = `
<DocumentFragment>
  @media screen and (min-width:480px) and (max-width:959px) {
  .emotion-0 {
    margin: 16px;
  }
}

@media screen and (min-width:960px) {
  .emotion-0 {
    margin: 8px;
  }
}

@media screen and (max-width:959px) {
  .emotion-0 {
    padding: 4px 8px;
  }
}

@media screen and (min-width:960px) {
  .emotion-0 {
    padding: 12px 16px;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`SpaceInset renders responsive override spaceInset010Squish for xs, spaceInset020Squish for sm, and spaceInset030Squish for md breakpoints 1`] = `
<DocumentFragment>
  @media screen and (min-width:480px) and (max-width:959px) {
  .emotion-0 {
    margin: 16px;
  }
}

@media screen and (min-width:960px) {
  .emotion-0 {
    margin: 8px;
  }
}

@media screen and (max-width:479px) {
  .emotion-0 {
    padding: 4px 8px;
  }
}

@media screen and (min-width:480px) and (max-width:959px) {
  .emotion-0 {
    padding: 8px 12px;
  }
}

@media screen and (min-width:960px) {
  .emotion-0 {
    padding: 12px 16px;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`SpaceInset renders responsive override spaceInset010Squish for xs, spaceInset020Squish for sm, and spaceInset030Squish for md breakpoints in reverser order 1`] = `
<DocumentFragment>
  @media screen and (min-width:480px) and (max-width:959px) {
  .emotion-0 {
    margin: 16px;
  }
}

@media screen and (min-width:960px) {
  .emotion-0 {
    margin: 8px;
  }
}

@media screen and (max-width:479px) {
  .emotion-0 {
    padding: 4px 8px;
  }
}

@media screen and (min-width:480px) and (max-width:959px) {
  .emotion-0 {
    padding: 8px 12px;
  }
}

@media screen and (min-width:960px) {
  .emotion-0 {
    padding: 12px 16px;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`SpaceInset renders responsive override spaceInset010Squish for xs, spaceInset020Squish for sm, and wrong for md breakpoints 1`] = `
<DocumentFragment>
  @media screen and (min-width:480px) and (max-width:959px) {
  .emotion-0 {
    margin: 16px;
  }
}

@media screen and (min-width:960px) {
  .emotion-0 {
    margin: 8px;
  }
}

@media screen and (max-width:479px) {
  .emotion-0 {
    padding: 4px 8px;
  }
}

@media screen and (min-width:480px) {
  .emotion-0 {
    padding: 8px 12px;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`SpaceInset renders responsive override spaceInset010Squish for xs, spaceInset020Squish for sm, spaceInset030Squish for md, spaceInset040Squish for lg breakpoints and ignore wrong prop 1`] = `
<DocumentFragment>
  @media screen and (min-width:480px) and (max-width:959px) {
  .emotion-0 {
    margin: 16px;
  }
}

@media screen and (min-width:960px) {
  .emotion-0 {
    margin: 8px;
  }
}

@media screen and (max-width:479px) {
  .emotion-0 {
    padding: 4px 8px;
  }
}

@media screen and (min-width:480px) and (max-width:959px) {
  .emotion-0 {
    padding: 8px 12px;
  }
}

@media screen and (min-width:960px) and (max-width:1439px) {
  .emotion-0 {
    padding: 12px 16px;
  }
}

@media screen and (min-width:1440px) {
  .emotion-0 {
    padding: 16px 24px;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`SpaceInset renders with override incorrect object 1`] = `
<DocumentFragment>
  @media screen and (min-width:480px) and (max-width:959px) {
  .emotion-0 {
    margin: 16px;
  }
}

@media screen and (min-width:960px) {
  .emotion-0 {
    margin: 8px;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`SpaceInset renders with override spaceInset010Squish 1`] = `
<DocumentFragment>
  .emotion-0 {
  padding: 4px 8px;
}

@media screen and (min-width:480px) and (max-width:959px) {
  .emotion-0 {
    margin: 16px;
  }
}

@media screen and (min-width:960px) {
  .emotion-0 {
    margin: 8px;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;
